lxc/completion.go

```diff
@@ -372,8 +372,10 @@ func (g *cmdGlobal) cmpInstanceSetKeys(instanceName string) ([]string, cobra.She
 
 	for configKey := range instance.Config {
 		// We only want to return the intersection between allInstanceConfigKeys and configKeys to avoid returning the full instance config.
+		// We also need to account for user defined keys such as user.* and linux.sysctl.*.
 		_, exists := keySet[configKey]
-		if exists {
+		if exists || strings.HasPrefix(configKey, "user.") ||
+			strings.HasPrefix(configKey, "linux.sysctl.") {
```

Comment:
Searching for `.*` on https://documentation.ubuntu.com/lxd/en/latest/reference/instance_options/ I think only the `environment.` prefix is missing from your extended list.